Origin: https://github.com/openssl/openssl/commit/0daccd4dc1f1ac62181738a91714f35472e50f3c
Reviewed-by: Sylvain Beucler <beuc@debian.org>
Last-Update: 2021-09-13

commit 0daccd4dc1f1ac62181738a91714f35472e50f3c
Author: Viktor Dukhovni <openssl-users@dukhovni.org>
Date:   Thu Jan 28 03:01:45 2016 -0500

    Check chain extensions also for trusted certificates
    
    This includes basic constraints, key usages, issuer EKUs and auxiliary
    trust OIDs (given a trust suitably related to the intended purpose).
    
    Added tests and updated documentation.
    
    Reviewed-by: Dr. Stephen Henson <steve@openssl.org>

Index: openssl1.0-1.0.2u/crypto/x509/x509_vpm.c
===================================================================
--- openssl1.0-1.0.2u.orig/crypto/x509/x509_vpm.c
+++ openssl1.0-1.0.2u/crypto/x509/x509_vpm.c
@@ -537,7 +537,7 @@ static const X509_VERIFY_PARAM default_t
      "default",                 /* X509 default parameters */
      0,                         /* Check time */
      0,                         /* internal flags */
-     0,                         /* flags */
+     X509_V_FLAG_TRUSTED_FIRST, /* flags */
      0,                         /* purpose */
      0,                         /* trust */
      100,                       /* depth */
